src/main/java/mate/academy/spring/security/CustomUserDetailService.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.spring.security;
+
+import java.util.Optional;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailService implements UserDetailsService {
+    @Autowired
+    private UserService userService;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+
+        Optional<User> userOptional = userService.findByEmail(email);
+        if (userOptional.isPresent()) {
```

Comment:
```suggestion
        Optional<User> userOptional = userService.findByEmail(email).orElseThrow(
                        () -> new UsernameNotFoundException("Can't find user: " + username));
```
this way is preferable (in real projects as well) but that is just a suggestion